src/libs/actions/Report.js

```diff
@@ -323,6 +333,81 @@ function fetchChatReportsByIDs(chatList) {
         });
 }
 
+/**
+ * Given IOU and chat report ID fetches most recent IOU data from API and update stored data in Onyx.
+ *
+ * @param {Number} iouReportID
+ * @param {Number} chatReportID
+ */
+function updateIOUReportData(iouReportID, chatReportID) {
+    API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: iouReportID,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    }).then((response) => {
+        if (!response) {
+            return;
+        }
+
+        if (response.jsonCode !== 200) {
+            console.error(response.message);
+            return;
+        }
+
+        const iouReportData = response.reports[iouReportID];
+        if (!iouReportData) {
+            console.error(`No iouReportData found for reportID ${iouReportID}`);
+            return;
+        }
+        const iouReportObject = getSimplifiedIOUReport(iouReportData, chatReportID);
+        const chatReportObject = {
+            hasOutstandingIOU: iouReportObject.stateNum === 1
+                && iouReportObject.total !== 0,
+            iouReportID,
+        };
+        if (!chatReportObject.hasOutstandingIOU) {
+            chatReportObject.iouReportID = null;
+        }
+        const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+        const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`;
+        Onyx.merge(iouReportKey, iouReportObject);
+        Onyx.merge(reportKey, chatReportObject);
+    });
+}
+
+/**
+ * Given chat report ID finds active IOU report ID via GetIOUReport API call
+ *
+ * @param {Number} chatReportID
+ * @returns {Promise}
+ */
+function fetchIOUReportID(chatReportID) {
+    const chatReport = lodashGet(allReports, chatReportID);
+    const participants = chatReport.participants;
+    if (participants.length !== 1) {
+        console.error('chatReport doesn\'t have any participant.');
+        return;
+    }
+    return API.GetIOUReport({
+        debtorEmail: participants[0],
+    }).then((response) => {
+        const iouReportID = response.reportID || 0;
+        if (response.jsonCode !== 200) {
+            console.error(response.message);
+            return;
+        }
+        if (iouReportID === 0) {
+            // If there is no IOU report for this user then we will assume it has been paid and do nothing here.
+            // All reports are initialized with hasOutstandingIOU: false. Since the IOU report we were looking for has
+            // been settled then there's nothing more to do.
+            console.debug('GetIOUReport returned a reportID of 0, not fetching IOU report data');
+            return;
+        }
+        return iouReportID;
```

Comment:
Ok now that we have this can we use it to replace these lines? The code is very similar if not the same?

https://github.com/Expensify/Expensify.cash/blob/f64e2f3b95ba15666f6963ae42cce2fe537e2719/src/libs/actions/Report.js#L235-L249